heimdall-common/src/main/java/pm/gnosis/heimdall/common/utils/ObservableUtils.kt

```diff
@@ -22,6 +22,9 @@ fun <D> Observable<out Result<D>>.subscribeForResult(onNext: ((D) -> Unit)?, onE
             onError?.invoke(it)
         })
 
+fun <D> Observable<out Result<D>>.doOnNextForResult(onNext: ((D) -> Unit)?, onError: ((Throwable) -> Unit)? = null): Observable<out Result<D>> =
```

Comment:
I think we are creating extensions for specific scenarios. Sometimes it is more clear to have those steps inline.